department-of-veterans-affairs/abd-vro
# upgrade to gradle 8

## What was the problem?
deprecated gradle features were being used making VRO incompatible with the latest Gradle version

Associated tickets or Slack threads:

https://github.com/orgs/department-of-veterans-affairs/projects/871/views/7?pane=issue&itemId=27157250

## How does this fix it?[^1]
upgrades gradle plugins that were using deprecated Gradle features
adds explicit dependencies where necessary
changes Gradle property names where necessary

## How to test this PR
- `./gradlew build check test`
- `./gradlew dockerComposeUp`
- `./gradlew  build -x test -x jacocoTestCoverageVerification`
[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
